pageserver/src/tenant_mgr.rs

```diff
@@ -108,6 +108,10 @@ pub fn init_tenant_mgr(
 /// Attempts to load as many entites as possible: if a certain timeline fails during the load, the tenant is marked as "Broken",
 /// and the load continues.
 ///
+/// For successful tenant attach, it first has to have a `timelines/` subdirectory and a tenant config file that's loaded into memory successfully.
+/// If either of the conditions fails, the tenant will be added to memory with [`TenantState::Broken`] state, otherwise we start to load its timelines.
```

Comment:
> Do we have any way to fix broken timelines now?

2.5 of them actually:
* the most commonly used one is supposed to be the one you were describing below, from pageserver's remote storage
* if we corrupt / loose the storage somehow, we're able to retract to safepeekers' data, either inmem (for the recent data) or the S3 one (semi manual process)
* nothing prevents us from adding walbackup handling and have another source of restoration, if I understand the mechanism right, hence 0.5 points

Reply:
> Will the flow be

I think, now there's no automated flow for this particular situation.
We have 
* "delete timeline" operation that's supposed to remove files from memory, local fs and remote storage, as the name implies
* "detach tenant" that removes the tenant and all of its timelines from the memory and from local FS, not from the remote storage. 
* plan to add "reload" tenant for unloading the data from memory only: https://github.com/neondatabase/neon/issues/2537

Either of the operations could be used to restore timeline data, the simplest restoration source would be remote storage, albeit it's not reliable yet, since has not finished CRC integration.
One could retrieve the correct layer files from the remote storage and try restarting pageserver or reloading its tenant, or do the reattach to use remote storage data.

IMO, we might benefit from per-timeline operations for maintenance, but so far there's no RFC or discussion on the topic.

Reply:
Looks like SK restoration is more convenient and reliable (since WAL segments have CRC): https://github.com/neondatabase/cloud/wiki/Recovery-from-WAL